app/controllers/review_bids_controller.rb

```diff
@@ -40,36 +32,51 @@ def index
     @assignment = @participant.assignment
-    @sign_up_topics = SignUpTopic.where(assignment_id: @assignment.id, private_to: nil)
-    my_topic = SignedUpTeam.topic_id(@participant.parent_id, @participant.user_id)
-    @sign_up_topics -= SignUpTopic.where(assignment_id: @assignment.id, id: my_topic)
-    @num_participants = AssignmentParticipant.where(parent_id: @assignment.id).count
-    @selected_topics = nil # this is used to list the topics assigned to review. (ie select == assigned i believe)
-    @bids = ReviewBid.where(participant_id: @participant, assignment_id: @assignment.id)
+    @selected_topics= nil
+    topic_ids_with_team = SignedUpTeam.where.not(team_id: nil).pluck(:topic_id) #Topics which have been selected by teams for submission
+    @signup_topics = SignUpTopic.where(assignment_id: @assignment.id, id: topic_ids_with_team) #signup_topics is all the topics students have signed up for.
+    #remove own topic from set of topics to bid on 
+    my_topic = SignedUpTeam.topic_id(@participant.parent_id, @participant.user_id) 
+    @signup_topics -= SignUpTopic.where(assignment_id: @assignment.id, id: my_topic) # remove own topic from set of topics to bid on
+    
+ 
+    @num_participants = AssignmentParticipant.where(parent_id: @assignment.id).count  # gotta know # participants to determine if topic's hot
+    @assigned_topics= nil
+    # Create an instance of ReviewBid
+    @bids = ReviewBid.where(participant_id:@participant,assignment_id:@assignment.id)  # Update bids to be the list of sign-up topics on which the participant has bid
     signed_up_topics = []
     @bids.each do |bid|
-      sign_up_topic = SignUpTopic.find_by(id: bid.signuptopic_id)
-      signed_up_topics << sign_up_topic if sign_up_topic
+      signup_topic = SignUpTopic.find_by(id: bid.signuptopic_id)
+      signed_up_topics << signup_topic if signup_topic
     end
-    signed_up_topics &= @sign_up_topics
-    @sign_up_topics -= signed_up_topics
+    signed_up_topics &= @signup_topics #signed_up_topics is the set of topics students DID sign up for.  
+    @signup_topics -= signed_up_topics
     @bids = signed_up_topics
-    @num_of_topics = @sign_up_topics.size
-    @assigned_review_maps = []
-    ReviewResponseMap.where(reviewed_object_id: @assignment.id, reviewer_id: @participant.id).each do |review_map|
+    @num_of_topics = @signup_topics.size # count the remaining sign-up topics
+    @assigned_review_maps = []   #fetch review maps for the participant in the current assignment
+    #assigned_topics=[]
+    ReviewResponseMap.where({:reviewed_object_id => @assignment.id, :reviewer_id => @participant.id}).each do |review_map|
       @assigned_review_maps << review_map
-    end
-
-    # explicitly render view since it's in the sign up sheet views
+    end 
     render 'sign_up_sheet/review_bids_show'
   end
-
   # function that assigns and updates priorities for review bids
   def set_priority
+    # Create an instance of ReviewBid
```

Comment:
If it creates a bid, why does it end with "destroy_all"?  That needs to be explained.